src/Std/Init/Nat.lean

```diff
@@ -0,0 +1,46 @@
+/-
+Copyright (c) 2024 Lean FRO, LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Markus Himmel
+-/
+prelude
+import Init.Data.Nat.Bitwise.Lemmas
+import Init.RCases
+
+namespace Nat
+
+theorem lt_of_eq_of_lt {n m k : Nat} : n = m → m < k → n < k :=
+  fun h₁ h₂ => h₁ ▸ h₂
+
+theorem le_of_testBit {n m : Nat} (h : ∀ i, n.testBit i = true → m.testBit i = true) : n ≤ m := by
```

Comment:
I guess we need to work out where the split is between `Init` and `Std`, but this can be adjusted later.

Reply:
Sorry, I should have made this more clear: my plan is to move everything in the `Std/Init` folder into `Init` as part of separate PRs that are to be merged before this one. The only reason why I haven't done this yet is that I wanted the PR up by the end of the week and I ran out of week.